batch.py

```diff
@@ -0,0 +1,761 @@
+
+
+def get_promoter_variants(
+    mt_path: str,  # checkpoint from function above
+    ht_path: str,
+    gene_file: str,
+    gene_name: str,
+    window_size: int,
+    # plink_output_prefix: str, # figure out how to deal with this
+):
+    """Subset hail matrix table
+
+    Input:
+    mt_path: path to already subsetted hail matrix table
+    gene_file: path to a tsv file with information on
+    the chrom, start and end location of genes (rows)
+
+    Output:
+    For retained variants, that are: 1) in promoter regions and
+    2) within 50kb up or down-stream of the gene body (or in the gene body itself)
+    (on top of all filters done above)
+
+    returns:
+    - Path to plink prefix for genotype files (plink format: .bed, .bim, .fam)
+    - Path to hail table with variant (rows) stats for downstream analyses
+    """
+    # read hail matrix table object (pre-filtered)
+    init_batch()
+    mt = hl.read_matrix_table(mt_path)
+
+    # get relevant chromosome
+    gene_df = pd.read_csv(AnyPath(dataset_path(gene_file)), sep="\t", index_col=0)
+    chrom = gene_df[gene_df["gene_name"] == gene_name]["chr"]
+    # subset to chromosome
+    mt = mt.filter_rows(mt.locus.contig == ("chr" + chrom))
+
+    # subset to window
+    # get gene body position (start and end) and build interval
+    interval_start = int(gene_df[gene_df["gene_name"] == gene_name]["start"]) - int(
+        window_size
+    )
+    interval_end = int(gene_df[gene_df["gene_name"] == gene_name]["end"]) + int(
+        window_size
+    )
+    # clip to chromosome boundaries
+    left_boundary = max(1, interval_start)
+    right_boundary = min(
+        interval_end, hl.get_reference("GRCh38").lengths[f"chr{chrom}"]
+    )
+    # get gene-specific genomic interval
+    gene_interval = f"chr{chrom}:{left_boundary}-{right_boundary}"
+    logging.info(f"Interval considered: {gene_interval}")  # 'chr22:23219960-23348287'
+
+    # include variants up to {window size} up- and downstream
+    mt = hl.filter_intervals(
+        mt, [hl.parse_locus_interval(gene_interval, reference_genome="GRCh38")]
+    )
+    mt_path = output_path("in_window.mt", "tmp")
+    mt = mt.checkpoint(
+        mt_path, overwrite=True
+    )  # add checkpoint to avoid repeat evaluation
+    logging.info(f"Number of variants within interval: {mt.count()[0]}")
+
+    # annotate using VEP
+    vep_ht = hl.read_table(ht_path)
+    mt = mt.annotate_rows(vep=vep_ht[mt.row_key].vep)
+    mt_path = output_path("vep_annotated.mt", "tmp")
+    mt = mt.checkpoint(
+        mt_path, overwrite=True
+    )  # add checkpoint to avoid repeat evaluation
+    logging.info(f"Number of QC-passing, biallelic SNPs: {mt.count()[0]}")
+
+    # filter variants found to be in promoter regions
+    mt = mt.filter_rows(
+        mt.vep.regulatory_feature_consequences["biotype"].contains("promoter")
+    )
+    mt_path = output_path("promoter_variants.mt", "tmp")
+    mt = mt.checkpoint(mt_path, overwrite=True)  # checkpoint
```

Comment:
```suggestion
    promoter_path = output_path("promoter_variants.mt", "tmp")
    mt = mt.checkpoint(promoter_path, overwrite=True)  # checkpoint
```

Not a functional change, but as the path is different to the original `mt_path` this will help the code readability